### DISCUSSION TITLE
motion detection + recording time

### DISCUSSION THREAD

**Discussion by Fougl** [gopro/labs#909] [Q&A]
Hi, I would appreciate a solution where I use the following motion detection !SM5D4!5N!R

I would like to only have the trigger by motion detection, record for 10 seconds and stop recording, sleep for 5 seconds, regardless of movement. then again motion detection mode. I am missing only the second part, so record for 10 seconds. 

Cheers

**Comment by dnewman-gpsw** [comment]
Can't use the old !SMx method as you have custom rules.  You need to use the new scripting and test the m (motion) variable.
Something like:
```<r0>m2!S+!8E!5R```

**Comment by Fougl** [comment]
thanks, doesnt seem to work properly on hero12. just keeps starting recording even when i put m50 or sth

For example this just starts recoding everytime it does the repeat
![image](https://github.com/gopro/labs/assets/21178929/4b7578cb-6769-482f-bbe9-bbb5c9abd909)


Also, simply this command >m25!S+60E!R
It doesnt even trigger anything. only thing i can see is labs processing sign at the top, but nothing happens

Also, i am using a mask.pgm file . how does that work with the m type of motion detection?

**Reply by dnewman-gpsw** [reply]
>m50 or >m25 would almost never trigger as that means 50% (or 25%) of pixels are changing significantly  >m1 or >m2 is typical for a trigger.  You can even do >m0.5 etc.

However, masking doesn't work with the scripting.  So you can't do what you want.  

P.S.  +60E is not a valid command, that should have been +!60E

**Reply by Fougl** [reply]
The camera just keep getting trigered recording. That is my point. There is no motion but recordings are being made constantly. Like in a loop every time repeats. Regardless of the m percantage. I dont think it works properly. I think you should test it

Command >m5!S+!1E!+!2N!R just keeps getting triggered 

https://github.com/gopro/labs/assets/21178929/a39136a7-34d6-4e71-ba86-2a5b27534c93

![image](https://github.com/gopro/labs/assets/21178929/57d5de99-8fa4-495e-8644-044eab3eddc4)


With m25, if i put my hand in front of the camera, would not that mean that more than 25% of the pixels change? How many pixel percantage change does sensitivity 9 in the old motion detection have for example?

So no plans of incorporating the mask.pgm part? Nobody finds that part useful? Cause it really is super useful for many commercial applications. And in my particular case of recording customers, the mask part comes in super useful. And with old command it works almost without false positives, or false negatives. The only thing I am missing to be super flexible and keep the recording times to minimum is this kind of conditions. is there no easy way to develop the mask option with the new command as well?

Here is one of my requests in open gopro https://github.com/gopro/OpenGoPro/issues/431
![image](https://github.com/gopro/labs/assets/21178929/b4c4c42f-f67e-4219-af90-828cb2bba7e2)

So you can see, how this things are super useful

**Reply by Fougl** [reply]
@dnewman-gpsw did you have a chance to have a look?

**Reply by dnewman-gpsw** [reply]
At Cannes Film Festival for the rest of week.  I will test when I get back.

**Reply by Fougl** [reply]
Amazing! Enjoy it. Thanks for the reply

**Reply by dnewman-gpsw** [reply]
Yep, I have a film in the Short Film Corner. Fantastic to be here.

**Comment by dnewman-gpsw** [comment]
I can reproduce the problem.  When the motion detection is getting motion noise in the frame when the capture stops, causing a trigger for the next capture.  Looks like a fix is needed.
